### PR TITLE
ADXL362 add separate triggers for activity/inactivity

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds separate triggers for activity and inactivity detection
for the ADXL362 accelerometer.
Previously, applications could not distinguish these events.

Signed-off-by: Maximilian Deubel [maximilian.deubel@nordicsemi.no](mailto:maximilian.deubel@nordicsemi.no)

### CODE REVIEW

**Comment by carlescufi** [comment]
@gmarull @avisconti could you please take a look?

**Review by nvlsianpu — DISMISSED** [review]
Code looks fine to me.
Need to add records to the release note: one for ADXL362 and second note for sensor API extension.

File: include/zephyr/drivers/sensor.h [hunk 1]
```diff
@@ -236,6 +236,11 @@ enum sensor_trigger_type {
 	/** Trigger fires when a free fall is detected. */
 	SENSOR_TRIG_FREEFALL,
 
+	/** Trigger fires when motion is detected. */
+	SENSOR_TRIG_INSTANT_MOTION,
```
Comment (henrikbrixandersen) [inline include/zephyr/drivers/sensor.h:240]: Why `INSTANT_`?

**Review by henrikbrixandersen — DISMISSED** [review]

File: samples/sensor/adxl362/src/main.c [hunk 2]
```diff
@@ -22,8 +22,11 @@ static void trigger_handler(const struct device *dev,
 		}
 		k_sem_give(&sem);
 		break;
-	case SENSOR_TRIG_THRESHOLD:
-		printf("Threshold trigger\n");
+	case SENSOR_TRIG_INSTANT_MOTION:
+		printf("Activity trigger\n");
+		break;
+	case SENSOR_TRIG_STATIONARY:
+		printf("Inactivity trigger\n");
```
Comment (MaureenHelm) [inline samples/sensor/adxl362/src/main.c:29]: Please align the printf strings to match the names of the triggers

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: include/zephyr/drivers/sensor.h (hunk 1 above)
Comment (maxd-nordic) [reply include/zephyr/drivers/sensor.h]: done

File: drivers/sensor/adxl362/adxl362.h [hunk 3]
```diff
@@ -196,8 +196,10 @@ struct adxl362_data {
 	struct gpio_callback gpio_cb;
 	struct k_mutex trigger_mutex;
 
-	sensor_trigger_handler_t th_handler;
-	struct sensor_trigger th_trigger;
+	sensor_trigger_handler_t inact_th_handler;
+	struct sensor_trigger inact_th_trigger;
+	sensor_trigger_handler_t act_th_handler;
+	struct sensor_trigger act_th_trigger;
```
[the remark this answers is not in the document]
Comment (maxd-nordic) [reply drivers/sensor/adxl362/adxl362.h]: done

File: samples/sensor/adxl362/src/main.c (hunk 2 above)
Comment (maxd-nordic) [reply samples/sensor/adxl362/src/main.c]: done

**Review by nvlsianpu — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**